client/xdag_config.c

```diff
@@ -450,59 +444,60 @@ static void* xdag_config_open(const char *path)
 	return (void*)cfg;
 }
 
-
 static void xdag_config_close(void *cfg)
 {
 	CFG *config = (CFG *)cfg;
 	xdag_config_free(config);
 }
 
-
-
-int get_pool_config(const char *path, char *buf, int buflen)
+int parse_section(void *cfg, const char *section, static char **keys, int keys_count, char *buffer)
 {
-
-	char *key[9] = { "ip","port","max_connection_count_input", "max_miner_ip_count","connections_per_miner_limit","pool_fee","pool_reward","pool_direct","pool_fund" };
-	if(path) {
-		void *cfg = xdag_config_open(path);
-
-		if(cfg == NULL) {
+	for(int i = 0; i < keys_count; ++i) {
+		const char *value = xdag_config_get_value(cfg, section, keys[i], "");
+		if(isEmpty(value)) {
+			printf("Configuration parameter %s:%s cannot be empty.\n", section, keys[i]);
 			return -1;
 		}
-		const char *str = xdag_config_get_value(cfg, "POOL", "ip", "");
-
-		for(int i = 0; i < 9; i++) {
-			int len = 0;
-			str = xdag_config_get_value(cfg, "POOL", key[i], "");
-			if(strlen(str) == 0) {
-				printf("Configuration parameter cannot be empty\n");
-				return -1;
-			}
-			if(i == 0) {
-				len = strlen(str) + 1;
-			} else {
-				len = strlen(str) + strlen(buf) + 1;
-			}
-			if(len > buflen) {
-				printf("Configuration parameter error\n");
-				return -1;
-			}
-			if(i == 0) {
-				strcpy(buf, str);
-			} else {
-				strcat(buf, str);
-			}
-			if(i < 9 - 1) {
-				strcat(buf, ":");
-			}
 
+		if(strlen(buffer) + strlen(value) + 2 >= CONFIG_BUG_LENGTH) {
+			printf("Wrong configuration.\n");
+			return -1;
 		}
-		xdag_config_close(cfg);
+		if(i > 0) {
+			strcat(buffer, ":");
+		}
+		strcat(buffer, value);
 	}
-
-	return 0;
```

Comment:
why no return?

Reply:
You right, I missed return in the parse_section function.

Reply:
Fixed